File: src/main/java/ai/ancf/lmos/operator/server/routing/CustomResourcesService.java
/*
 * SPDX-FileCopyrightText: 2024 Deutsche Telekom AG
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package ai.ancf.lmos.operator.server.routing;

import ai.ancf.lmos.operator.resources.channel.ChannelResource;
import ai.ancf.lmos.operator.resources.routing.ChannelRoutingResource;

import java.util.List;
import java.util.Optional;

public interface CustomResourcesService {

    Optional<ChannelRoutingResource> getRouting(String tenant, String channel, String subset);

    List<ChannelResource> getChannels(String tenant, String subset);

    Optional<ChannelResource> getChannel(String tenant, String channel, String subset);
}

File: src/main/kotlin/ai/ancf/lmos/operator/server/routing/CustomResourcesService.kt
/*
 * SPDX-FileCopyrightText: 2024 Deutsche Telekom AG
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package ai.ancf.lmos.operator.server.routing

import ai.ancf.lmos.operator.resources.ChannelResource
import ai.ancf.lmos.operator.resources.ChannelRoutingResource

interface CustomResourcesService {
    fun getRouting(
        tenant: String,
        channel: String,
        subset: String,
        namespace: String,
    ): ChannelRoutingResource?

    fun getChannels(
        tenant: String,
        subset: String,
        namespace: String,
    ): List<ChannelResource>

    fun getChannel(
        tenant: String,
        channel: String,
        subset: String,
        namespace: String,
    ): ChannelResource?
}
